src/auth/baseexternalclient.ts

```diff
@@ -127,6 +128,8 @@ export abstract class BaseExternalAccountClient extends AuthClient {
   private readonly subjectTokenType: string;
   private readonly serviceAccountImpersonationUrl?: string;
   private readonly stsCredential: sts.StsCredentials;
+  private readonly clientAuth: ClientAuthentication | undefined;
+  private readonly workforcePoolUserProject: string | undefined;
```

Comment:
Note that `getProjectId()` should use this to determine project ID:
`async getProjectId(): Promise<string | null> {`

`GoogleAuth` relies on this.

See python implementation: https://github.com/googleapis/google-auth-library-python/pull/868/files